px-checkout/src/main/java/com/mercadopago/android/px/internal/datasource/PaymentMethodRepositoryImpl.kt

```diff
@@ -14,5 +14,8 @@ internal class PaymentMethodRepositoryImpl(private val fileManager: FileManager)
 
     override fun readFromStorage() = fileManager.readAnyList(file, PaymentMethod::class.java)
 
-    override fun getPaymentMethodById(paymentMethodId: String) = value.firstOrNull { it.id == paymentMethodId }
+    override fun getPaymentMethodById(paymentMethodId: String): PaymentMethod {
+        return value.firstOrNull { it.id == paymentMethodId } ?: throw IllegalStateException("PaymentTypeId null")
```

Comment:
@samantaalbanez I updated this, can you test the crash with this update?

Also can we update the IllegalStateException to

```suggestion
        return value.firstOrNull { it.id == paymentMethodId } ?: throw IllegalStateException("PaymentMethod not foundl")
```